CMakeLists.txt

```diff
@@ -1422,14 +1418,15 @@ status("    Build with moment combine:"  ENABLE_MOMENTS_COMBINE_MATRICES THEN "y
 # ===================== Optional 3rd parties =====================
 status("")
 status("  Mathematics: ")
-status("    Use MKL:"                USE_MKL            THEN "yes" ELSE "no")
-status("    Use OpenBLAS:"           USE_OPENBLAS       THEN "yes" ELSE "no")
-status("    Use Atlas:"              USE_ATLAS          THEN "yes" ELSE "no")
-status("    Use Netlib Lapack:"      USE_LAPACK_NETLIB  THEN "yes (ver ${LAPACK_C_VERSION})" ELSE "no")
-status("    Use Lapack (built-in):"  WITH_LAPACK        THEN "yes (ver ${LAPACK_VERSION})" ELSE "no")
+status("    Blas/Lapack:"            (USE_LAPACK OR WITH_LAPACK) THEN "yes" ELSE "no")
+status("    \\- Use MKL:"            USE_MKL            THEN "yes" ELSE "no")
+status("    \\- Use OpenBLAS:"       USE_OPENBLAS       THEN "yes" ELSE "no")
+status("    \\- Use Atlas:"          USE_ATLAS          THEN "yes" ELSE "no")
+status("    \\- Use Netlib:"         USE_LAPACK_NETLIB  THEN "yes (ver ${LAPACK_C_VERSION})" ELSE "no")
```

Comment:
I can do it, but from the tests presented in `tutorial-basic-linear-algebra.doc` in Lapack performance comparison table, `Netlib` sounds faster than `GSL`.  That's why here the order of the 3rd parties is the performance order (1. `MKL`, 2. `OpenBlas`, 3. `Atlas`, 4. `Netlib`, 5. `GSL`, 6. `built_in`).

You don't agree with that choice ?

Reply:
I have some doubt with `update-alternatives --config libblas.so.3-<multiarch>`. I think somehow it does not work as intended. When Atlas and OpenBLAS are available, ViSP will be linked to `libopenblas.so.0`. So changing the alternative would not work.

Reply:
So, I think your results with Atlas and Netlib are wrong due to this issue.

Netlib is the reference Fortran Lapack code. There is no optimization and the benchmark results should be more or less equivalent between Netlib and Lapack built-in (minor the perf difference between the two languages and the compiler optimizations).

---

For instance, your results are:

| Operation | OpenBLAS | Atlas | Netlib | Built-in | Naive
| --- | --- | --- | --- | --- | --- |
| (600x400)x(400x600) | 3.19575 ms | 3.9736 ms | 3.61614 ms | 170.769 ms | 165.217 ms
| (400x600)x(600x400) | 2.01295 ms | 1.82563 ms | 1.79074 ms | 102.953 ms | 105.26 ms

Mine are:

| Operation | OpenBLAS | Atlas | Netlib | Built-in | Naive
| --- | --- | --- | --- | --- | --- |
| (600x400)x(400x600) | 1.43821 ms | 19.5031 ms | 60.6548 ms | 144.417 ms | 146.238 ms
| (400x600)x(600x400) | 985.012 us | 12.9778 ms | 40.6998 ms | 98.5517 ms | 99.2755 ms

I am using Docker with different images where only Atlas, OpenBLAS or Netlib are installed.

Reply:
I would put GSL at the same level than Eigen. [GSL](https://www.gnu.org/software/gsl/) does more than just providing the Lapack routines. And it can use a 3rd party BLAS library: [Using the GSL with an optimized BLAS](https://wiki.debian.org/DebianScience/LinearAlgebraLibraries).

Reply:
We cannot put GSL at the same level as Eigen because of the incompatibility between GSL and MKL (see issue #735) that was originally the reason of this PR. There should be a mutual exclusion between GSL and MKL. Even if GSL can do more than Lapack, my choice to consider GSL as a Lapack 3rdparty was motivated by the fact that in ViSP GSl is used to perform similar linear algebra operation than MKL, OpenBLAS, Atlas, Netlib